app/src/main/java/com/github/polypoly/app/menu/JoinGameLobbyActivity.kt

```diff
@@ -537,20 +537,8 @@ class JoinGameLobbyActivity : MenuActivity("Join Game") {
     private fun joinGameLobbyRoom() {
         val currentLobbyKey = DB_GAME_LOBIES_PATH + gameLobbyCode
         remoteDB.getValue<GameLobby>(currentLobbyKey).thenAccept { gameLobby ->
-            val newGameLobby = GameLobby(
-                gameLobby.admin,
-                gameLobby.gameMode,
-                gameLobby.minimumNumberOfPlayers, gameLobby.maximumNumberOfPlayers, gameLobby.roundDuration
-                , gameLobby.gameMap, gameLobby.initialPlayerBalance, gameLobby.name, gameLobby.code, gameLobby.private)
-
-            for (player in gameLobby.usersRegistered) {
-                if (player != newGameLobby.admin) {
-                    newGameLobby.addUser(player)
-                }
-            }
-
-            newGameLobby.addUser(fakeAuthenticatedUser)
-            remoteDB.updateValue(currentLobbyKey, newGameLobby)
+            gameLobby.addUser(fakeAuthenticatedUser)
+            remoteDB.updateValue(currentLobbyKey, gameLobby)
```

Comment:
Perfect! You don't even need to update the DB as this will be done in `GameLobby` ;)